ecosystem/sep-0038.md

```diff
@@ -0,0 +1,176 @@
+## Preamble
+
+```
+SEP: 0038
+Title: Request for Quote API
+Author: Jake Urban <@jakeurban>
+Track: Standard
+Status: Draft
+Created: 2021-04-09
+Discussion: https://github.com/stellar/stellar-protocol/issues/901
+Version 1.0.0
+```
+
+## Summary
+
+This protocol enables anchors to accept off-chain assets in exchange for different on-chain assets, and vice versa. Specifically, it enables anchors to provide quotes that can referenced within the context of existing Stellar Ecosystem Proposals. How the exchange of assets is facilitated is outside the scope of this document.
+
+## Motivation
+
+Anchoring an asset and issuing an asset are distinct functions that have different buisness and technical requirements. However, issuing an asset has often been a prerequisite for anchoring an asset. This protocol enables anchors to transfer value on and off the Stellar network for their clients regardless of whether a one-for-one reserve-backed Stellar asset existed for the off-chain asset held by the anchor's clients.
+
+Removing this requirement for anchors also provides downstream benefits to ecosystem participants generally. Enabling anchors to accept any Stellar asset will naturally decrease liquidity fragmentation on the decentralized exchange, leading to greater market depth and tigher spreads between trading pairs.
+
+## Specification
+
+### Authentication
+
+All endpoints require authentication in the form of a [SEP-10](sep-0010.md) JSON Web Token (JWT) in the `Authorization` header of the request. Authenticated accounts must exist on the network in order to make requests to `POST /quote`, as described in the [Protecting Against Bad Actors](#protecting-against-bad-actors) section.
+
+### Content Type
+
+All endpoints accept in requests the following `Content-Type`:
+- `application/json`
+
+All endpoints respond with content type:
+- `application/json`
+
+### Endpoints
+
+* [`GET /info`](#get-info)
+* [`GET /prices`](#get-prices)
+* [`GET /price`](#get-price)
+* [`POST /quote`](#post-quote)
+
+### GET Info
+
+This endpoint describes the supported Stellar assets and off-chain currencies available for trading. Note that the anchor may not support a trading pair between every Stellar asset and off-chain currency listed. Use the [`GET /prices`](#get-prices) endpoint to see which pairs are supported.
+
+#### Request
+
+No request arguments required.
+
+#### Response
+
+Name | Type | Description
+-----|------|------------
+`assets` | array | An array of objects describing the Stellar assets available in exchange for one or more of the `currencies` listed.
+`currencies` | array | An array of objects describing the off-chain currencies available in exchange for one or more of the `assets` listed.
+
+`assets` Object Schema
+
+Name | Type | Description
+-----|------|------------
+`asset` | string | The [SEP-32](sep-0032.md) address of the Stellar asset. Lumens should be specified as `XLM`.
```

Comment:
❓ The way we've been standardizing lumens is to refer to it as `native`. That's the SEP-11 form and I believe some of our other APIs use this form, although ticker has used native as the issuer account instead, essentially `XLM:native`. Motivation for using only `XLM` here?

SEP-32 isn't adopted yet, and I need to do some work on it. For now I'd prefer it if you could run with the SEP-11 asset form, `<code>:<issuer-account>`, and then let's modify this to state SEP-32 or SEP-11 once SEP-32 is solid.

If you would prefer SEP-32, I can put some energy into firming it up quickly so it is ready to go for this.

Reply:
Admittedly I initially thought SEP-32 was the standardization of the `<code>:<issuer-account>` syntax and decided to stick with the protocol anyway since I assumed you were trying to move the ecosystem in that direction.

I can change both points 👍

Reply:
On second thought I don't think SEP-32 is appropriate here. SEP-32 is targeting Uuser facing interfaces and should almost never be used in programmatic interfaces. A client should use SEP-32 to resolve an asset address to a code and issuer then use the code and issuer with any API and services it interacts. This is because a client should rey only on itself to resolve and verify an issuer account and asset address domain match.